From b3ae3efa9aa4f819ea52eebea27b615d78d396b2 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 28 Nov 2018 13:36:46 +0200
Subject: [PATCH 5229/5242] MLK-20361: drm/panel/raydium: increase vsync_len
 to 4

commit  a49ffcd97cd8203b583ec5b1ce2224090efd573d from
https://source.codeaurora.org/external/imx/linux-imx.git

This is a work-around for the shifted display issue when using DCSS with
the DDR self-refresh disabled.

Also, make sure it works with LCDIF as well.

Increasing VSYNC length gives DCSS more time to fetch the data from RAM
before the next frame kicks in.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
(cherry picked from commit 3c53532c466add234ac3ca2e70cefe80ea308b4f)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/panel/panel-raydium-rm67191.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/panel/panel-raydium-rm67191.c b/drivers/gpu/drm/panel/panel-raydium-rm67191.c
index 940f780..647ba3f 100644
--- a/drivers/gpu/drm/panel/panel-raydium-rm67191.c
+++ b/drivers/gpu/drm/panel/panel-raydium-rm67191.c
@@ -528,7 +528,7 @@ static int rad_bl_update_status(struct backlight_device *bl)
 	.hback_porch = { 34, 34, 34 },
 	.vactive = { 1920, 1920, 1920 },
 	.vfront_porch = { 10, 10, 10 },
-	.vsync_len = { 2, 2, 2 },
+	.vsync_len = { 4, 4, 4 },
 	.vback_porch = { 4, 4, 4 },
 	.flags = DISPLAY_FLAGS_HSYNC_LOW |
 		 DISPLAY_FLAGS_VSYNC_LOW |
-- 
1.7.9.5

